kafka/consumer.go

```diff
@@ -180,8 +180,11 @@ func (c *Consumer) fetch(ctx context.Context) error {
 	fetches := c.client.PollRecords(ctx, c.cfg.MaxPollRecords)
 	defer c.client.AllowRebalance()
 
-	if fetches.IsClientClosed() || errors.Is(fetches.Err0(), context.Canceled) {
-		return context.Canceled // Client closed or context cancelled.
+	if fetches.IsClientClosed() {
+		return fmt.Errorf("client is closed: %w", context.Canceled)
+	}
+	if errors.Is(fetches.Err0(), context.Canceled) {
+		return fmt.Errorf("context canceled: %w", fetches.Err0())
 	}
 	switch c.cfg.Delivery {
 	case apmqueue.AtLeastOnceDeliveryType:
```

Comment:
**question**: we are not checking for error when committing uncommited offsets. Should we log an error or/and return the error to the caller ?

Reply:
No need, since this function is running on a loop. 

If the offsets are committed back to Kafka, the events will be re-processed once the processing timeout is hit, adhering to the desired at least once delivery.

Reply:
Oh I see, thanks for explanation! :bow: